### PR TITLE
[EOC] Add readable values and update functions

### PR DESCRIPTION
**Pull request by lispcoc**
#### Summary
None

#### Purpose of change
Add readable values to EoC and update functions.

#### Describe the solution
New readable value
- u_val('weight') - Character/Monster/Item weight. (mg)
- u_val('volume') - Monster/Item volume. (ml) Doesn't work for characters.

New math function
- u_skill_exp('skill_id') - Get/Set skill experience.

Updated function
- npc_die - Make it possible to destroy monsters or items.

#### Testing
Add example codes (example_eocs.json), works collectly.

### CODE REVIEW

**Comment by PatrikLundell** [comment]
Minor comment on the wording of the PR: the second point about volume still says "weight" and then (correctly) expresses it in ml.

**Review by andrei8l — COMMENTED** [review]
Math part looks good; thanks. Just have a couple of minor things.

File: src/math_parser_diag.cpp [hunk 1]
```diff
@@ -364,6 +364,50 @@ std::function<void( dialogue &, double )> skill_ass( char scope,
...
+    } else {
+        if( params.size() > 1 ) {
+            raw_value = params[1];
+        }
+    }
+
+    return [skill_value, raw_value, beta = is_beta( scope ) ]( dialogue const & d,
+    double val ) {
+        skill_id skill( skill_value.str( d ) );
+        bool raw = raw_value.str( d ) == "true";
```
[the remark this answers is not in the document]
Comment (lispcoc) [reply src/math_parser_diag.cpp]: Fixed, thx!

**Comment by lispcoc** [comment]
All tests passed(last error is caused by random mapgen?)

**Pull request merged**